internal/graph/reachableresources.go

```diff
@@ -226,31 +191,114 @@ func (crr *ConcurrentReachableResources) lookupRelationEntrypoint(ctx context.Co
 		}
 		defer it.Close()
 
-		for tpl := it.Next(); tpl != nil; tpl = it.Next() {
+		return crr.chunkedRedispatch(it, func(resourceIdsFound []string) error {
+			return crr.redispatchOrReport(ctx, &core.RelationReference{
+				Namespace: relationReference.Namespace,
+				Relation:  relationReference.Relation,
+			}, resourceIdsFound, rg, g, entrypoint, stream, req, dispatched)
+		})
+	})
+
+	return nil
+}
+
+func (crr *ConcurrentReachableResources) chunkedRedispatch(it datastore.RelationshipIterator, handler func(resourceIdsFound []string) error) error {
+	chunkIndex := 0
+
+	for {
```

Comment:
On first glance I immediately thought this would exceed the length of `dispatchChunkSizes` before I saw the compensation code in the loop body. Can this be rewritten maybe to use a monotonically increasing `chunkIndex` and then use a size function or something?

Reply:
Now it can be re-written as a for loop.

Reply:
I'm not a huge fan of using the syntax for loops that break using internal conditions, but I changed it and added a comment to indicate that